datalake/src/main/java/com/sequenceiq/datalake/service/FreeipaService.java

```diff
@@ -36,12 +51,38 @@ public void checkFreeipaRunning(String envCrn) {
 
     }
 
-    DescribeFreeIpaResponse describe(String envCrn) {
+    public DescribeFreeIpaResponse describe(String envCrn) {
         try {
             return freeIpaV1Endpoint.describe(envCrn);
         } catch (NotFoundException e) {
             LOGGER.error("Could not find freeipa with envCrn: " + envCrn, e);
         }
         return null;
     }
+
+    public long getNodeCount(String envCrn) {
+        return describe(envCrn)
+                .getInstanceGroups().stream().mapToLong(InstanceGroupBase::getNodeCount).sum();
+    }
+
+    public String upscale(String envCrn) {
```

Comment:
Please update the method to be generic by adding FormFactor as an argument. Please pass FormFactor.HA as an argument while invoking from FreeIpaUpscaleActions.